expo/gbe/views/review_costume_view.py

```diff
@@ -1,118 +1,56 @@
-                            urlconf='gbe.urls',
-                            args=[costume_id])
-    else:
-            actionform = False
-            actionURL = False
+        self.readonlyform_pieces = [
+            self.object_form,
+            self.details,
+            self.performer,
+            self.profile]
 
-    '''
-    if user has previously reviewed the act, provide their review for update
-    '''
-    try:
-        bid_eval = BidEvaluation.objects.filter(
-            bid_id=costume_id,
-            evaluator_id=reviewer.resourceitem_id)[0]
-    except:
-        bid_eval = BidEvaluation(evaluator=reviewer, bid=costume)
+    def get(self, request, *args, **kwargs):
+        self.groundwork(request, args, kwargs)
+        self.form = BidEvaluationForm(instance=self.bid_eval)
+        return (self.object_not_current_redirect() or
+                self.bid_review_response(request))
 
-    # show costume info and inputs for review
-    if request.method == 'POST':
-        form = BidEvaluationForm(request.POST, instance=bid_eval)
-        if form.is_valid():
-            evaluation = form.save(commit=False)
-            evaluation.evaluator = reviewer
-            evaluation.bid = costume
-            evaluation.save()
-            return HttpResponseRedirect(reverse('costume_review_list',
-                                                urlconf='gbe.urls'))
-        else:
-            return render(request, 'gbe/bid_review.tmpl',
-                          {'readonlyform': [
-                                costume_form,
-                                details,
-                                performer,
-                                profile],
-                           'reviewer': reviewer,
-                           'form': form,
-                           'actionform': actionform,
-                           'actionURL': actionURL,
-                           'conference': conference,
-                           'old_bid': old_bid,
-                           })
-    else:
-        form = BidEvaluationForm(instance=bid_eval)
-        return render(request,
-                      'gbe/bid_review.tmpl',
-                      {'readonlyform': [
-                            costume_form,
-                            details,
-                            performer,
-                            profile],
-                       'reviewer': reviewer,
-                       'form': form,
-                       'actionform': actionform,
-                       'actionURL': actionURL,
-                       'conference': conference,
-                       'old_bid': old_bid,
-                       })
+    def post(self, request, *args, **kwargs):
+        self.groundwork(request, args, kwargs)
+        self.form = BidEvaluationForm(request.POST, instance=self.bid_eval)
+        return (self.object_not_current_redirect() or
+                self.post_response_for_form(request))
```

Comment:
So... the get and post seem to be repeated from child class to child class.  Is there a reason why we can't put the default get/post into the parent?